sdktests/server_side_persistence_read_write.go

```diff
@@ -0,0 +1,469 @@
+package sdktests
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/launchdarkly/go-sdk-common/v3/ldcontext"
+	"github.com/launchdarkly/go-sdk-common/v3/ldvalue"
+	m "github.com/launchdarkly/go-test-helpers/v2/matchers"
+	h "github.com/launchdarkly/sdk-test-harness/v2/framework/helpers"
+	"github.com/launchdarkly/sdk-test-harness/v2/framework/ldtest"
+	o "github.com/launchdarkly/sdk-test-harness/v2/framework/opt"
+	"github.com/launchdarkly/sdk-test-harness/v2/mockld"
+	"github.com/launchdarkly/sdk-test-harness/v2/servicedef"
+	"github.com/stretchr/testify/require"
+)
+
+func (s *ServerSidePersistentTests) doReadWriteTests(t *ldtest.T) {
+	// No cache is enabled
+	t.Run("initializes store when data received", s.initializesStoreWhenDataReceived)
+	t.Run("applies updates to store", s.appliesUpdatesToStore)
+
+	t.Run("data source updates respect versioning", s.dataSourceUpdatesRespectVersioning)
+	t.Run("data source deletions respect versioning", s.dataSourceDeletesRespectVersioning)
+
+	cacheConfigs := []servicedef.SDKConfigPersistentCache{
+		{Mode: servicedef.Infinite},
+		{Mode: servicedef.TTL, TTL: o.Some(1)},
+	}
+
+	for _, cacheConfig := range cacheConfigs {
+		t.Run(fmt.Sprintf("cache mode %s", cacheConfig.Mode), func(t *ldtest.T) {
+			t.Run("does not cache flag miss", func(t *ldtest.T) {
+				s.doesNotCacheFlagMiss(t, cacheConfig)
+			})
+			t.Run("sdk reflects data source updates even with cache", func(t *ldtest.T) {
+				s.sdkReflectsDataSourceUpdatesEvenWithCache(t, cacheConfig)
+			})
+			t.Run("ignores dropped flags", func(t *ldtest.T) {
+				s.ignoresDroppedFlagsWithForeverCache(t)
+			})
+		})
+	}
+
+	t.Run("infinite cache", func(t *ldtest.T) {
+		t.Run("ignores dropped flags", s.ignoresDroppedFlagsWithForeverCache)
+	})
+
+	t.Run("ttl cache", func(t *ldtest.T) {
+		t.Run("ignores dropped flags", s.ignoresDroppedFlagsWithTTLCache)
+	})
+}
+
+func (s *ServerSidePersistentTests) initializesStoreWhenDataReceived(t *ldtest.T) {
+	require.NoError(t, s.persistentStore.Reset())
+
+	persistence := NewPersistence()
+	persistence.SetStore(servicedef.SDKConfigPersistentStore{
+		Type: s.persistentStore.Type(),
+		DSN:  s.persistentStore.DSN(),
+	})
+	persistence.SetCache(servicedef.SDKConfigPersistentCache{
+		Mode: servicedef.Off,
```

Comment:
I'm wondering if we'll quickly run into some other constant named `Off`. Might be worth prefixing it with `CacheMode` so `CacheModeOff`.